lib/pbench/server/api/resources/users_api.py

```diff
@@ -326,36 +370,24 @@ def get(self, username):
                     }
         """
         try:
-            user, verified = self.auth.verify_user(username)
-        except Exception:
-            self.logger.exception("Exception occurred during verifying the user")
-            abort(500, message="INTERNAL ERROR")
-
-        # TODO: Check if the user has the right privileges
-        if verified:
-            response_object = user.get_json()
-            return make_response(jsonify(response_object), 200)
-        elif user.is_admin():
-            try:
-                target_user = User.query(username=username)
-                response_object = target_user.get_json()
-                return make_response(jsonify(response_object), 200)
-            except Exception:
-                self.logger.exception(
-                    "Exception occurred while querying the user. Username: {}", username
-                )
-                abort(500, message="INTERNAL ERROR")
-        else:
-            self.logger.warning(
-                "User {} is not authorized to get user {}.", user.username, username,
+            target_user, http_status, http_message = self.get_valid_target_user(
+                target_username, "GET"
             )
-            abort(
-                403,
-                message=f"Not authorized to access information about user {username}",
+        except Exception:
+            self.logger.exception(
+                "Exception occurred in GET request from user {!r} while validating target user {!r}",
+                self.auth.token_auth.current_user().username,
+                target_username,
             )
+            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
+        if target_user:
+            response_object = target_user.get_json()
+            return make_response(jsonify(response_object), HTTPStatus.OK)
+        else:
+            abort(http_status, message=http_message)
```

Comment:
This is a nit, but....  The other two HTTP methods do a negative check with an abort and otherwise perform the request...it would be nice to match that structure here:
```
        if not target_user:
            abort(http_status, message=http_message)

        response_object = target_user.get_json()
        return make_response(jsonify(response_object), HTTPStatus.OK)
```